src/twisted/internet/defer.py

```diff
@@ -1966,16 +2173,15 @@ def __init__(self, name, scheduler=None):
         if scheduler is None:
             from twisted.internet import reactor
 
-            scheduler = reactor
+            scheduler = cast(IReactorTime, reactor)
```

Comment:
Hmm. This `cast` seems like it's a problem, but probably not *this* PR's problem.

Reply:
The problem here is that mypy thinks the reactor is `ModuleType`.  It should be fixed in another PR, though I'm not sure what more module voodoo we can do about it.
My guess is that it would really be better if the normal way to get the installed reactor was via a callable and not a weird module import that's really a reactor.

Reply:
Yeah I'd be very happy if Mypy just encouraged us to get rid of the gross `from twisted.internet import reactor` idiom entirely.